freqtrade/utils.py

```diff
@@ -21,6 +23,7 @@
 from freqtrade.exchange import (available_exchanges, ccxt_exchanges,
                                 market_is_active, symbol_is_pair)
 from freqtrade.misc import plural, render_template
+from freqtrade.optimize.hyperopt import Hyperopt
```

Comment:
don't ... (just don't).

With this, the whole bot will require the hyperopt dependencies again - which does not make sense (and in addition to that, causes the test-failure since its creating a cyclic import).

we have 2 options for this:
* move this import into each function which needs it, 
* move the new "hyperopt" commands to `optimize/__init__.py`. This is where the backtesting / hyperopt commands are located. (note: if we don't opt for this approach, we should consider moving `start_backtesting()` and `start_hyperopt()` into this file.

Reply:
Agree (no dependency of hyperopt stuff for the whole freqtrade)

Initially I made those methods as functions outside of the Hyperopt class, but moved them back to their initial place as static methods to simplify reviewing the changes. As functions, they can easily be placed into a separate module (like `hyperopt_misc.py`) which will remove all the dependencies from other hyperopt stuff...

Reply:
you'll still need to move the import into the function itself so it's only imported when they are actually called - since the functions themselfs are imported into arguments.

Why movethem to a new new file though - `freqtrade/optimize/__init__.py` should work just fine (?)

Reply:
I do not like placing anything except imports from submodules into `__init__.py` files and defining this bad design.

Reply:
hm, then we should move backtesting / hyperopt out of that too... especially the hyperopt one goes "hand in hand" with this PR ...

Reply:
my other idea was to create a `commands` or `subcommands` dir and move every `start_*` function into a separate file (module) in it... Because `utils.py` gets bigger and bigger...